src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Reload/ProjectReloadManager.cs

```diff
@@ -0,0 +1,560 @@
+                            }
+                            break;
+                        }
+                    }
+                }
+                else
+                {
+                    if(ignoreAll)
+                    {   
+                        // do nothing
+                    }
+                    else if(reloadAll)
+                    {
+                        ReloadProjectInSolution(failure.Item1);
+                    }
+                    else
+                    {
+
+                        var buttons = new string[] {Resources.IgnoreAll,
+                                                    Resources.Ignore,
+                                                    Resources.ReloadAll,
+                                                    Resources.Reload};
+
+                        var msgText = string.Format(Resources.ProjectModificationsPrompt, Path.GetFileNameWithoutExtension(failure.Item1.ProjectFile));
+                        switch (_dialogServices.ShowMultiChoiceMsgBox(Resources.ProjectModificationDlgTitle, msgText, buttons))
+                        {
+                            case MultiChoiceMsgBoxResult.Cancel:
+                            {
+                                break;
+                            }
+                            case MultiChoiceMsgBoxResult.Button1:
+                            {
+                                // Ignore All
+                                ignoreAll = true;
+                                break;
+                            }
+                            case MultiChoiceMsgBoxResult.Button2:
+                            {
+                                // Ignore
+                                break;
+                            }
+                            case MultiChoiceMsgBoxResult.Button3:
+                            {
+                                // Reload  all
+                                reloadAll = true;
+                                ReloadProjectInSolution(failure.Item1);
+                                break;
+                            }
+                            case MultiChoiceMsgBoxResult.Button4:
+                            {
+                                // Reload
+                                ReloadProjectInSolution(failure.Item1);
+                                break;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Helper to use the solution to reload the project.
+        /// Reloading is managed via the ReloadItem() method of our parent hierarchy (solution 
+        /// or solution folder).  So first we get our parent hierarchy and our itemid in the parent
+        /// hierarchy. 
+        /// </summary>
+        void ReloadProjectInSolution(IReloadableProject project)
+        {
+            // Get our parent hierarchy and our itemid in the parent hierarchy.
```

Comment:
Any reason we can't use CPS's version of this? ie  UnconfiguedProject.ReloadAsync